include/zephyr/bluetooth/encrypted_ad_data.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+
+#define BT_EAD_AAD_SIZE	       1  /**< Additional Authenticated Data size in bytes */
+#define BT_EAD_RANDOMIZER_SIZE 5  /**< Randomizer size in bytes */
+#define BT_EAD_NONCE_SIZE      13 /**< nonce size in bytes */
+#define BT_EAD_KEY_SIZE	       16 /**< Key size in bytes */
+#define BT_EAD_IV_SIZE	       8  /**< Initialisation Vector size in bytes */
+#define BT_EAD_MIC_SIZE	       4  /**< MIC size in bytes */
+
+/* This value is used to set the directionBit of the CCM  nonce to the MSB of the Randomizer field
+ * (see Supplement to the Bluetooth Core   Specification v11, Part A 1.23.3)
+ */
+#define BT_EAD_RANDOMIZER_DIRECTION_BIT 7
+
+/* Fixed value used for the Additional Authenticated Data (see Supplement to the Bluetooth Core
+ * Specification v11, Part A 1.23.3)
+ */
+static uint8_t bt_ead_aad[] = {0xEA};
+BUILD_ASSERT(sizeof(bt_ead_aad) == BT_EAD_AAD_SIZE);
```

Comment:
With your solution the array can be empty. With the build assert we ensure that the array is exactly the size we expect it to be.

Reply:
What do you mean by `array can be empty.`? 

It can only be empty if `BT_EAD_AAD_SIZE`

Reply:
If the size of the array is specified in the declaration, it's not possible to check if there are fewer elements than expected. Any elements missing from the initializer will be static-initialized. For example:
```c
static uint8_t bt_ead_aad[BT_EAD_AAD_SIZE] = {};
/* Legal according to standard. No warnings. `bt_ead_aad[0]` is guaranteed to be `0`. */
/* Even, worse: The following check now passes.
 * For the check to work as expected, the size of the array must be implicitly
 * specified by the number of elements in the initializer, i.e. remove BT_EAD_AAD_SIZE
 * from the definition of bt_ead_aad. */
BUILD_ASSERT(sizeof(bt_ead_aad) == BT_EAD_AAD_SIZE);
```

Reply:
> /* Even, worse: The following check now passes. */

Which is why it can be removed. 

The `BUILD_ASSERT(sizeof(bt_ead_aad) == BT_EAD_AAD_SIZE);` doesn't verify correctness in the first place; it just verifies the size. If we want to ensure the size, then we can just use my suggestion which ensures size correctness (without the build assert), and while
```c
static uint8_t bt_ead_aad[BT_EAD_AAD_SIZE] = {};
```
would jus have `bt_ead_add[0] == 0`, it would give a build error if we attempted to do
```c
static uint8_t bt_ead_aad[BT_EAD_AAD_SIZE] = {0xEA, 0xEB};
```

My point was that the BUILD_ASSERT's purpose can be done in a simpler way (IMO)

Reply:
Hmm, I see your point that in the case of the array being too small, it would be a difference, and we could forget to ensure the values. 

My suggestion can be ignored.